### PR TITLE
fix(eslint-plugin): [no-unsafe-enum-comparison] exempt bit shift operators

### LINKED ISSUE #7067: Bug: no-unsafe-enum-comparison - False positives with bit shifts
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Repro Code

```typescript
enum Foo {
  Bar,
}

const myBitShift = 1 << Foo.Bar
```

This gives the following error:

> The two values in this comparison do not have a shared enum type.

The error message says that we are doing a "comparison", but this is false. We are not doing a comparison at all, we are doing a bit shift.

Thus, I propose that the `<<` operator and the `>>` operator are exempt from being examined by this rule, and I assume that this was just an oversight when the rule was originally implemented. (Well, I was actually the person to originally implement it, but I guess this behavior got lost in translation somewhere.)

<br>

### Playground

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=KYOwrgtgBAYg9nKBvAsAKClAQgQwE4A06AvuugMZwgDOALlBAJ5YCWtAygBYsBm9AvFACMUADyjYCAHS486IA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZonpNYAW1pkOY4nyrIOTdFETRoHaJHBgAviB1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### ESLint Config

I am using the recommend configs, i.e.:

```
  extends: [
    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
    "plugin:@typescript-eslint/recommended",

    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended-requiring-type-checking.ts
    "plugin:@typescript-eslint/recommended-requiring-type-checking",

    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/strict.ts
    "plugin:@typescript-eslint/strict",
  ],
```

### PR DESCRIPTION
**Pull request by Melandra**
## PR Checklist

- [x] Addresses an existing open issue: fixes #7067 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Exempts the bit shift operators `<<` and `>>` from the `no-unsafe-enum-comparison` rule.

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts [hunk 1]
```diff
@@ -66,6 +66,19 @@ export default util.createRule({
...
+        // Exempt bit shift operators since they are not comparisons
+        //
+        // ```ts
+        // enum Fruit {
+        //  Apple,
+        // }
+        //
+        // const myBitShift = 1 << Fruit.Apple;
+        // ```
+        if (node.operator === '>>' || node.operator === '<<') {
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts:78]: [Suggestion] We already have some exclusions in the in the `'BinaryExpression[operator=/=|<|>/]'` [esquery](https://github.com/estools/esquery). How about adding a `!=` to exclude these? `[operator=/.../][operator!=/<<|>>/]`?

There's definitely a way to get this working in just one regexp if you really want. 😄

File: packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts [hunk 2]
```diff
@@ -66,6 +66,19 @@ export default util.createRule({
...
       ): void {
+        // Exempt bit shift operators since they are not comparisons
+        //
+        // ```ts
+        // enum Fruit {
+        //  Apple,
+        // }
+        //
+        // const myBitShift = 1 << Fruit.Apple;
+        // ```
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts:77]: [Nit] I think this comment is pretty clear from the code & tests. WDYT?

```suggestion
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
LGTM, thanks for sending! ✨ 

Just a couple of non-blocking suggestions - what do you think?

File: packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts [hunk 3]
```diff
@@ -66,6 +66,19 @@ export default util.createRule({
       'BinaryExpression[operator=/=|<|>/]'(
```
Comment (Josh-Cena) [inline packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts:66]:
```suggestion
      'BinaryExpression[operator=/^[=<>]$/]'(
```

...Much simpler?

Comment (Melandra) [reply packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts]: Thanks for the suggestion, please see [3d812c6](https://github.com/typescript-eslint/typescript-eslint/pull/7074/commits/3d812c69909794f8ede28c717cf86fbae5cabf63), I've replaced the regex with `^[<>!]?={0,3}$` instead to also match `!==` and `===`

File: packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts (hunk 2 above)
Comment (Melandra) [reply packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts]: Please see [3d812c6](https://github.com/typescript-eslint/typescript-eslint/pull/7074/commits/3d812c69909794f8ede28c717cf86fbae5cabf63) where I've removed the if statement along with the comment

File: packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts (hunk 1 above)
Comment (Melandra) [reply packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts]: Please see [3d812c6](https://github.com/typescript-eslint/typescript-eslint/pull/7074/commits/3d812c69909794f8ede28c717cf86fbae5cabf63).

**Comment by JoshuaKGoldberg** [comment]
@Melandra just checking in, did you mean to re-request our review on this one?

**Comment by Melandra** [comment]
> @Melandra just checking in, did you mean to re-request our review on this one?

Yes, I didn't know you were expecting me to re-request the review, sorry 😬

**Comment by JoshuaKGoldberg** [comment]
Ah we've since added guidance in https://typescript-eslint.io/contributing/pull-requests#addressing-feedback-and-beyond - which I think didn't exist when you first sent this. Thanks!

**Review by JoshuaKGoldberg — APPROVED** [review]
🔥 great, thanks!

Pam from The Office happily saying 'nice'

**Pull request merged**